### PR TITLE
Fix typo in transforming-telemetry.md and collector.md

### PR DESCRIPTION
**Pull request by ydFu**
The reasons for this correction are as follows:

- Correction of the word "kubernetes" from its original spelling mistake to the correct form.
- Rectification of some grammatical errors, such as the addition of periods or the removal of redundant words.

### CODE REVIEW

File: content/en/docs/collector/transforming-telemetry.md [hunk 1]
```diff
@@ -139,7 +139,7 @@ processors:
 **Processor**:
 [resource detection processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor)
 and
-[k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor)
+[k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor).
```
Comment (svrnm) [inline content/en/docs/collector/transforming-telemetry.md:142]:
```suggestion
[k8sattributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor)
```

File: content/en/docs/kubernetes/helm/collector.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: OpenTelemetry Collector Chart
 linkTitle: Collector Chart
 # prettier-ignore
-cSpell:ignore: filelog filelogreceiver filelogreceiver hostmetricsreceiver kubelet kubeletstats kubeletstatsreceiver kuberenetes loggingexporter otlphttp sattributesprocessor sclusterreceiver sobjectsreceiver statefulset
+cSpell:ignore: filelog filelogreceiver filelogreceiver hostmetricsreceiver kubelet kubeletstats kubeletstatsreceiver kubernetes loggingexporter otlphttp sattributesprocessor sclusterreceiver sobjectsreceiver statefulset
```
Comment (svrnm) [inline content/en/docs/kubernetes/helm/collector.md:5]:
```suggestion
cSpell:ignore: filelog filelogreceiver filelogreceiver hostmetricsreceiver kubelet kubeletstats kubeletstatsreceiver loggingexporter otlphttp sattributesprocessor sclusterreceiver sobjectsreceiver statefulset
```

**Review by svrnm — CHANGES_REQUESTED** [review]
Thanks @ydFu ! Please take a look at my suggestions

File: content/en/docs/collector/transforming-telemetry.md [hunk 3]
```diff
@@ -98,7 +98,7 @@ processors:
 ## Renaming Metrics or Metric Labels
 
 **Processor:**
-[metrics transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor)
+[metrics transform processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor).
```
[the remark this answers is not in the document]
Comment (ydFu) [reply content/en/docs/collector/transforming-telemetry.md]: done

File: content/en/docs/collector/transforming-telemetry.md (hunk 1 above)
Comment (ydFu) [reply content/en/docs/collector/transforming-telemetry.md]: done

File: content/en/docs/kubernetes/helm/collector.md (hunk 2 above)
Comment (ydFu) [reply content/en/docs/kubernetes/helm/collector.md]: done

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
Thank you for taking the time to fix this typos @ydFu !

**Pull request merged**